tools/cldr-code/src/main/java/org/unicode/cldr/tool/CLDRModify.java

```diff
@@ -1259,6 +1236,130 @@ public void handlePath(String xpath) {
             }
         });
 
+        // 'P' Process, like 'p' but without inheritance replacement
+        fixList.add('P', "input-Processor-no-inheritance-replacement", new CLDRFilter() {
+            private DisplayAndInputProcessor inputProcessor;
+
+            @Override
+            public void handleStart() {
+                inputProcessor = new DisplayAndInputProcessor(cldrFileToFilter, true);
+            }
+
+            @Override
+            public void handleEnd() {
+                inputProcessor = null; // clean up, just in case
+            }
+
+            @Override
+            public void handlePath(String xpath) {
+                String value = cldrFileToFilter.getStringValue(xpath);
+                String newValue = inputProcessor.processInput(xpath, value, null);
+                if (value.equals(newValue)) {
+                    return;
+                }
+                String fullXPath = cldrFileToFilter.getFullXPath(xpath);
+                replace(fullXPath, fullXPath, newValue);
+            }
+        });
+
+        // use DAIP for one thing only: replaceBaileyWithInheritanceMarker
+        fixList.add('I', "Inheritance-substitution", new CLDRFilter() {
+            private DisplayAndInputProcessor inputProcessor;
+            private final int STEPS_FROM_ROOT = 1; // only process if locale's level matches; root = 0, en = 1, ...
+
+            @Override
+            public void handleStart() {
+                int steps = stepsFromRoot(cldrFileToFilter.getLocaleID());
+                if (steps == STEPS_FROM_ROOT) {
+                    inputProcessor = new DisplayAndInputProcessor(cldrFileToFilter, true);
+                    inputProcessor.enableInheritanceReplacement(getResolved());
+                } else {
+                    inputProcessor = null;
+                }
+            }
+
+            private int stepsFromRoot(String origLoc) {
+                int steps = 0;
+                String loc = origLoc;
+                while (!LocaleNames.ROOT.equals(loc)) {
+                    loc = LocaleIDParser.getParent(loc);
+                    if (loc == null) {
+                        throw new IllegalArgumentException("Missing root in inheritance chain");
+                    }
+                    ++steps;
+                }
+                System.out.println("stepsFromRoot = " + steps + " for " + origLoc);
+                return steps;
+            }
+
+            @Override
+            public void handleEnd() {
+                inputProcessor = null; // clean up, just in case
+            }
+
+            @Override
+            public void handlePath(String xpath) {
+                if (inputProcessor == null) {
```

Comment:
probably want to throw exception here instead of just returning. That is, we sould always have an input processor for each locale.

Reply:
actually for -fU we get inputProcessor == null for a few locales including ajp that don't occur in the pre-drop data -- details are in one of the draft PRs

also we get inputProcessor == null if steps != STEPS_FROM_ROOT

these are cases where the current data will be kept without modification

Reply:
OK if this is a TODO - fix missing inputProcessor for some locales.